gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagActionReminderScheduler.java

```diff
@@ -192,4 +171,44 @@ public static Trigger createReminderJobTrigger(DagActionStore.LeaseParams leaseP
         .startAt(new Date(getCurrentTimeMillis.get() + reminderDurationMillis))
         .build();
   }
+
+  public class ReminderJobFactory implements JobFactory {
+    @Override
+    public Job newJob(TriggerFiredBundle bundle, Scheduler scheduler) {
+      return new ReminderJob();
+    }
+  }
+
+  /**
+   * Class used to store information regarding a pending dagAction that needs to be revisited at a later time
+   * by {@link DagManagement} interface to re-attempt a lease on if it has not been completed by the previous owner.
+   * These jobs are scheduled and used by the {@link DagActionReminderScheduler}.
+   */
+  public class ReminderJob implements Job {
```

Comment:
non-static inner classes are good for tightly-coupled abstractions.  as you just found, they have some pitfalls.  in this case, it feels more like a `static` inner class.  just give it a `private final DagManagement` member and `@RequiredArgsCtor`

the `ReminderJobFactory` would still be non-`static`.  it's the one that will take its `DagActionReminderScheduler.this.dagManagement` member and pass that along to the job's ctor.

Reply:
so you are saying `ReminderJobFactory` is more tightly coupled with `DagActionReminderScheduler` than `ReminderJob` is to `DagActionReminderScheduler` ?

Reply:
I'm not convinced the factory is actually that related to the `DagActionReminderScheduler`.  I was more accepting the compromise that it would be a way for the latter to "share" its `DagManagement` with the many `ReminderJob`s.  also there's only one factory.

but, TBH, it would be perfectly fine, and IMO preferable if the `ReminderJobFactory` were static and also held a `private final DagManagement` (w/ `@RequiredArgsConstructor`), so it could be created as:
```
    this.quartzScheduler.setJobFactory(new ReminderJobFactory(dagManagement));
```